src/hotspot/share/gc/shenandoah/shenandoahAllocRequest.hpp

```diff
@@ -167,71 +190,35 @@ class ShenandoahAllocRequest : StackObj {
   }
 
   inline bool is_mutator_alloc() const {
-    switch (_alloc_type) {
-      case _alloc_tlab:
-      case _alloc_shared:
-      case _alloc_cds:
-        return true;
-      case _alloc_gclab:
-      case _alloc_plab:
-      case _alloc_shared_gc:
-        return false;
-      default:
-        ShouldNotReachHere();
-        return false;
-    }
+    return (_alloc_type & bit_gc_alloc) == 0;
   }
 
   inline bool is_gc_alloc() const {
-    switch (_alloc_type) {
-      case _alloc_tlab:
-      case _alloc_shared:
-      case _alloc_cds:
-        return false;
-      case _alloc_gclab:
-      case _alloc_plab:
-      case _alloc_shared_gc:
-        return true;
-      default:
-        ShouldNotReachHere();
-        return false;
-    }
+    return (_alloc_type & bit_gc_alloc) != 0;
   }
 
   inline bool is_lab_alloc() const {
-    switch (_alloc_type) {
-      case _alloc_tlab:
-      case _alloc_gclab:
-      case _alloc_plab:
-        return true;
-      case _alloc_shared:
-      case _alloc_shared_gc:
-      case _alloc_cds:
-        return false;
-      default:
-        ShouldNotReachHere();
-        return false;
-    }
+    return (_alloc_type & bit_lab_alloc) != 0;
   }
 
   bool is_old() const {
```

Comment:
Should all of these be inline?  (Looks to me like the implementations are no larger than the code required to call and process result.)